vm/src/dictdatatype.rs

```diff
@@ -275,6 +284,20 @@ impl<T: Clone> Dict<T> {
         Ok(ret)
     }
 
+    pub fn get2<K: DictKey>(
+        &self,
+        other: &Self,
+        vm: &VirtualMachine,
+        key: &K,
+    ) -> PyResult<Option<T>> {
+        let hash = key.key_hash(vm)?;
+        if let Some(x) = self._get_inner(vm, key, hash)? {
+            Ok(Some(x))
+        } else {
+            other._get_inner(vm, key, hash)
+        }
```

Comment:
`get_inner` returns `Result<Option>`, I think if we wanted to use combinators it would be a bunch of `transpose()` calls so I figured this would be the cleanest way to do it.

Reply:
Oh, yep. I missed `Result`. But not sure about `transpose()` because Result is unwrappable.

How about this?
```rust
Ok(self._get_inner(vm, key, hash)?.or_else(|| other._get_inner(vm, key, hash)?))
```

Reply:
The `?` inside the or_else wouldn't return from the or function, it would return from the closure